### PR TITLE
Revamp Ide Testing DSL

### PR DESCRIPTION
**Pull request by i-walker**
Based on some feedback in testing. There is a new testing dsl in the Ide, 
which supports features described in the PluginScope `e.g.: IdeMetaPlugin`, which can then be used to describe tests. 
The data type IdeTest has now 3 fields the `code`, `test` to define a test and the expected result. 
The remaining dependencies to unfold that data structure into a test are now subject to functions and are not implicitly added. 
So we get the best out of both worlds, ascribing different interpreters for tests and a neat DSL without duplicating code.

### CODE REVIEW

File: idea-plugin/src/main/kotlin/arrow/meta/ide/testing/env/IdeTestInterpreter.kt [hunk 1]
```diff
@@ -1,90 +1,131 @@
...
+ * import arrow.meta.ide.testing.env.ideTest
+ * import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+ * import org.junit.Test
+ * //sampleStart
+ * class ExampleTest : IdeTestSetUp() {
+ *   @Test
+ *   fun `general test schema`(): Unit =
+ *     ideTest(
+ *       myFixture = myFixture, // the IntelliJ test environment spins up [myFixture] automatically at runtime.
+ *       ctx = MyIdePlugin() // add here your ide plugin, to get dependencies and created features in the scope of [test] and [result].
```
Comment (danimontoya) [inline idea-plugin/src/main/kotlin/arrow/meta/ide/testing/env/IdeTestInterpreter.kt:74]: nice!

File: idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/comprehensions/ComprehensionsTest.kt [hunk 2]
```diff
@@ -1,75 +1,75 @@
...
+            it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
+          }
+        ),
+        IdeTest(
+          code = ComprehensionsTestCode.code2,
+          test = { code, myFixture, _ ->
+            collectLM(code, myFixture, ArrowIcons.BIND)
+          },
+          result = resolvesWith("LineMarkerTest for two LM on untyped variables") {
+            it.takeIf { descriptor -> descriptor.lineMarker.size == 2 && descriptor.slowLM.isEmpty() }
```
Comment (raulraja) [inline idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/comprehensions/ComprehensionsTest.kt:34]: can these repeated expressions be refactored to a function?, they all seem to do the same except change the source code.

Comment (i-walker) [reply idea-plugin/src/test/kotlin/arrow/meta/ide/plugins/comprehensions/ComprehensionsTest.kt:34]: Yeah for sure, I'll add that to the new PR

**Comment by i-walker** [comment]
Yes, @danimontoya I agree. I'll open a new PR with the changed position.

**Comment by danimontoya** [comment]
> Yes, @danimontoya I agree. I'll open a new PR with the changed position.

Yeah, no problem.. whenever u want!! It is explained anyways 👍

**Pull request merged**